docs/learn/learn-comparison-ethereum-2.md

```diff
@@ -124,6 +124,31 @@ as long as the logic compiles to Wasm and the shard provides an "execute block"
+
+Examples of optimistic rollup solutions include [Optimisim](https://www.optimism.io/), [Arbitrum](https://bridge.arbitrum.io/) and [Unipig](https://unipig.exchange/welcome).
+
+### Zero-knowledge Rollups
+
+[Zero-knowledge proofs](https://en.wikipedia.org/wiki/Zero-knowledge_proof) can be used as a "proof" of valid transactions when batching, effectively reducing the transaction payload to be the size of a zk-proof. This, similar to the optimistic approach, allows transaction data to be abstracted away and reduced significantly in size before being committed to the main chain. Similar to most on-chain transactions and states, a ZK-rollup's state is represented as a [merkle tree](https://en.wikipedia.org/wiki/Merkle_tree), and the cryptographic hash of the tree's root(Merkle root) is what is stored on-chain. This allows for efficient batching, and tracking of changes to the transaction states. Any changes to the state will require the operator who initiated the changes to compute a new state root and commit it to the on-chain contract. If the contract verifies the new state as valid, then provided root hash becomes the new state root.
+
+#### EIP 4844
+
+"Introduce a new transaction format for “blob-carrying transactions” which contain a large amount of data that cannot be accessed by EVM execution, but whose commitment can be accessed. The format is intended to be fully compatible with the format that will be used in full sharding."
+
+Until the full spec of ETH 2.0 sharding is implemented, this stopgap introduces the "blob-transactions" transaction format, in the same way they would be designed with sharding, but without sharding those transactions. Blob-transactions are transactions that carry a large amount of data that cannot be accessed by the EVM, but whose commitment can be accessed. Compared to full sharding where the space allocated for this additional transaction data will be ~16MB, this EIP allows for a reduced cap corresponding to a target of ~1MB and a limit of ~2MB, which the rollup transactions can utilize. 
+
+### Rollups in Polkadot
+
+In Polkadot, the sharding model already exists as the Relay-chain being the beacon chain and Parachains being the shards. Parachains are most similar in implementation to an Optimistic rollup and in architecture to a ZK-rollup. Parachain logic already runs a validity proof. The proof (the approvals protocol) is interactive, unlike ZK-rollups, which are non-interactive. Additionally, unlike ZK-rollups, there are no difficulties in creating parachains with Turing-complete logic. This is a fundamental weakness of ZK rollups, as Turing completeness within ZK circuits is not easy.
```

Comment:
'Parachain logic already...' could be edited for clarity. style-wise I'd omit 'already'.

Would be nice to link here.

We should clarify in this section that Parachains are akin to rollups and could even be considered as a category of rollup.

Reply:
Not sure if there is good literature out there yet to link something here